cli/azd/pkg/ux/multi_select.go

```diff
@@ -0,0 +1,466 @@
+		}
+	}
+
+	if end < filteredOptionsCount {
+		if end >= 10 {
+			printer.Fprintf("%s ...\n", indent)
+		} else {
+			printer.Fprintf("%s  ...\n", indent)
+		}
+	}
+}
+
+func (p *MultiSelect) validate() {
+	p.hasValidationError = false
+	p.validationMessage = ""
+
+	if len(p.filteredChoices) == 0 {
+		p.hasValidationError = true
+		p.validationMessage = "No options found matching the filter"
+	} else if p.submitted && len(p.selectedChoices) == 0 {
+		p.hasValidationError = true
+		p.validationMessage = "At least one option must be selected"
+	}
+}
+
+func (p *MultiSelect) renderHint(printer Printer) {
+	// Hint
+	if p.showHelp && p.options.HelpMessage != "" {
+		printer.Fprintln()
+		printer.Fprintf(
+			color.HiMagentaString("  %s %s\n",
+				BoldString("Hint:"),
+				p.options.HelpMessage,
+			),
+		)
+	}
+}
+
+func (p *MultiSelect) renderValidation(printer Printer) {
+	// Validation error
+	if !p.showHelp && p.hasValidationError {
+		printer.Fprintln()
+		printer.Fprintln(color.YellowString("  %s", p.validationMessage))
+	}
+}
+
+func (p *MultiSelect) renderMessage(printer Printer) {
+	printer.Fprintf(color.CyanString("? "))
+
+	// Message
+	printer.Fprintf(BoldString("%s: ", p.options.Message))
+
+	// Cancelled
+	if p.cancelled {
+		printer.Fprintf(color.RedString("(Cancelled)"))
+	}
+
+	// Selected Value(s)
+	if !p.cancelled {
+		selectedChoices := p.sortSelectedChoices()
+		selectionValues := make([]string, len(selectedChoices))
+		for index, choice := range selectedChoices {
+			selectionValues[index] = choice.Value
+		}
+
+		rawValue := strings.Join(selectionValues, ", ")
+		printer.Fprintf(color.CyanString(rawValue))
+	}
+
+	printer.Fprintln()
+
+	// Filter
+	if !p.cancelled && !p.complete && *p.options.EnableFiltering {
+		printer.Fprintln()
+		printer.Fprintf("  Filter: ")
+
+		if p.filter == "" {
+			p.cursorPosition = Ptr(printer.CursorPosition())
+			printer.Fprintf(color.HiBlackString("Type to filter list"))
+		} else {
+			printer.Fprintf(p.filter)
+			p.cursorPosition = Ptr(printer.CursorPosition())
+		}
+
+		printer.Fprintln()
+		printer.Fprintln()
+	}
+}
+
+// Render renders the Select component.
+func (p *MultiSelect) Render(printer Printer) error {
+	if p.currentIndex == nil {
+		p.currentIndex = Ptr(0)
+	}
+
+	indent := "  "
+	p.renderMessage(printer)
+
+	if p.complete || p.cancelled {
+		return nil
+	}
+
+	p.applyFilter()
+	p.renderOptions(printer, indent)
+
+	p.validate()
+	p.renderValidation(printer)
+
+	p.renderHint(printer)
+
+	p.renderFooter(printer)
+
+	if p.cursorPosition != nil {
+		printer.SetCursorPosition(*p.cursorPosition)
+	}
+
+	return nil
+}
+
+func (p *MultiSelect) renderFooter(printer Printer) {
+	if p.cancelled || p.complete {
+		return
+	}
+
+	printer.Fprintln()
+	printer.Fprintln(color.HiBlackString("───────────────────────────────────"))
+	printer.Fprintln(color.HiBlackString("Use ↑/↓ to move, <space> to select"))
+	printer.Fprintln(color.HiBlackString("Use <enter> to submit, <?> for help"))
```

Comment:
@wbreza @SophCarp For the instructions text, suggest replacing symbols with words where we can to make it screen reader accessible. I believe this was one of the learnings from the [accessibility user study](https://microsoft-my.sharepoint.com/:p:/p/austinauth/EcEzJhCzWbJJsuNKrTtVZRwBdr3LirMZ3RDZSoHAA2qGcA?e=uw6Tej) from last year.
```suggestion
	printer.Fprintln(color.HiBlackString("Use arrow keys to move, use space to select"))
	printer.Fprintln(color.HiBlackString("Use enter to submit, type ? for help"))
```

Reply:
Also worth noting from the accessibility findings:

>Our wording of the instructions add confusion to the interaction (specifically "press enter to confirm"). Suggested fix: "Remove 'press enter to select' since that is just default terminal behavior"


https://microsoft-my.sharepoint.com/:p:/p/austinauth/EcEzJhCzWbJJsuNKrTtVZRwBdr3LirMZ3RDZSoHAA2qGcA?e=A0vvfM